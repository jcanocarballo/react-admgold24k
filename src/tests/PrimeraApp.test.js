import React from 'react'
// import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
const { default: PrimerApp } = require("../PrimerApp");
import {shallow} from 'enzyme'

describe('Pruebas de componente PrimeraApp', () => {
  
  // test('debe mostrar el mensaje "Hola, soy Goku" ', () => {
  //   const saludo = 'Hola, soy Goku';
  //   const {getByText} = render(<PrimerApp saludo={saludo}></PrimerApp>)

  //   expect(getByText(saludo)).toBeInTheDocument();
  // });
  test('debe de mostar <Primera /> correctamente', () => {
    const saludo = 'Hola, soy Goku';
    const wraper = shallow(<PrimerApp saludo={saludo}></PrimerApp>);
    expect(wraper).toMatchSnapshot();
  });

  test('debe de mostar el subtitulo en viado por props', () => {
    const saludo = 'Hola, soy Goku';
    const subtitulo = 'Soy un subtitulo'
    const wraper = shallow(
      <PrimerApp 
        saludo={saludo}
        subtitulo= {subtitulo}
      ></PrimerApp>
    );

    const textoParrafo = wraper.find('p').text();

    expect(textoParrafo).toBe(subtitulo);
    
  });
  
});