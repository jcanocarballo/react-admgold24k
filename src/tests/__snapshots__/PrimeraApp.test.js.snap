// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente PrimeraApp debe de mostar <Primera /> correctamente 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</div>
`;
